src/libmongoc/src/mongoc/mongoc-log-private.h

```diff
@@ -34,4 +34,14 @@ mongoc_log_trace_bytes (const char *domain, const uint8_t *_b, size_t _l);
 void
 mongoc_log_trace_iovec (const char *domain, const mongoc_iovec_t *_iov, size_t _iovcnt);
 
+BSON_BEGIN_DECLS
+#define STOP_LOGGING_CHECK                                          \
+   int stop_logging;                                                \
+   stop_logging = !gLogFunc;                                        \
+   stop_logging = stop_logging || !_mongoc_log_trace_is_enabled (); \
+   if (stop_logging) {                                              \
+      return;                                                       \
+   }
```

Comment:
Suggest removing `int stop_logging` to simplify. Wrap with `if (1) { ... } else (void) 0` so `STOP_LOGGING_CHECK;` not result in an empty statement (This pattern was adopted somewhat recently.  See [Handling control flow inside macros](https://gustedt.wordpress.com/2011/02/02/handling-control-flow-inside-macros/)):
```c
#define STOP_LOGGING_CHECK                                 \
   if (1) {                                                \
      if (!gLogFunc || !_mongoc_log_trace_is_enabled ()) { \
         return;                                           \
      }                                                    \
   } else                                                  \
      (void) 0
```